.github/actions/setup-rust/action.yml

```diff
@@ -1,35 +1,47 @@
 name: "Setup Rust"
 description: "Sets up the correct Rust version and caching via sccache and a GCP backend"
 inputs:
-  sccache_enabled:
-    description: "Enable or disable the sccache service"
+  cache_backend:
+    description: "Choose between sccache or github"
     required: false
-    default: 'true'
+    default: 'sccache'
   targets:
     description: "Additional targets to install"
     required: false
     default: ""
 runs:
   using: "composite"
   steps:
-    - if: ${{ inputs.sccache_enabled == 'true' }}
+
+    # Setup sccache, works well for most Rust components
+    - if: ${{ inputs.cache_backend == 'sccache' }}
       id: auth
       uses: google-github-actions/auth@v2
       with:
         workload_identity_provider: "projects/397012414171/locations/global/workloadIdentityPools/github-actions-pool/providers/github-actions"
         service_account: "github-actions@github-iam-387915.iam.gserviceaccount.com"
         export_environment_variables: true
         create_credentials_file: true
-    - if: ${{ inputs.sccache_enabled == 'true' }}
+    - if: ${{ inputs.cache_backend == 'sccache' }}
       run: |
         echo "SCCACHE_GCS_BUCKET=firezone-staging-sccache" >> $GITHUB_ENV
         echo "SCCACHE_GCS_RW_MODE=READ_WRITE" >> $GITHUB_ENV
       shell: bash
-    - if: ${{ inputs.sccache_enabled == 'true' }}
+    - if: ${{ inputs.cache_backend == 'sccache' }}
       uses: mozilla-actions/sccache-action@v0.0.4
-    - if: ${{ inputs.sccache_enabled }}
```

Comment:
@thomaseizinger I don't think this conditional was ever true so we may not have been actually using the sccache this whole time.

Reply:
I think it must be, otherwise the post-run for sccache should not show any compile requests, right?

Friday, 750 cache hits out of 1,000+ requests https://github.com/firezone/firezone/actions/runs/8116797107/job/22187597267#step:20:4

This PR, 750 cache hits out of 1,000+ requests https://github.com/firezone/firezone/actions/runs/8141116547/job/22247701765#step:20:4

We could merge this for consistency but I don't think it'll do anything.

Reply:
Yeah, I realized after opening this that it probably doesn't change anything. It does fix the logic bug though, so I'll merge it anyway.

Darn, was really looking forward to some speedups.

Reply:
I don't even remember making this input but I think GitHub actions is all JS so "truthy" should be good enough.

Reply:
Appears to have been added in https://github.com/firezone/firezone/commit/b1738bdd46d89fa10cbf97b3a6ef53dc7686bcb0.

I am a bit confused about this test bed. Are we actually using that?

Reply:
@thomaseizinger Soon™. The intention was to have it run smoke tests on bare metal for clients (Apple clients must be run on baremetal).

I can pull it out as a draft PR for the time being.